repl/scala-2.11/src/test/scala/org/apache/spark/repl/ReplSuite.scala

```diff
@@ -381,4 +381,13 @@ class ReplSuite extends SparkFunSuite {
     assertDoesNotContain("error:", output)
     assertDoesNotContain("Exception", output)
   }
+
+  test("define case class and create Dataset together") {
+    val output = runInterpreter("local-cluster[1,1,1024]",
+      """
+        |case class A(value: Int); sqlContext.createDataset(Seq(A(1)))(newProductEncoder[A])
```

Comment:
We should use `Seq(A(1)).toDS()` here, but it exposes another bug of our spark shell: we can't find implicit for `DatasetHolder`. Will file a JIRA later.